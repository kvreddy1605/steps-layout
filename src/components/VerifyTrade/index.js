import React from 'react';
import './index.scss';

const verifyTradeLabels = ["Description", "Ticker", "Firm/Branch", "Trade Date", "Trade Type", "Account Type", "Trade Amount", "Currrency", "PLID"];

const createLabels = (labels) => (
    labels.map((label, index)=> {
        return (<div className="label" key={index}>{label}</div>)
    })  
);
const createFund = (fund) => {
    const {description, ticker, firm, tradeDate, tradeType, accountType, tradeAmount, currrency, PLID, shareBalance} = fund;
    return (<div className="fund">
                <div className="fund__row1">
                    <div>{description}</div>
                    <div>{ticker}</div>
                    <div>{firm}</div>
                    <div>{tradeDate}</div>
                    <div>{tradeType}</div>
                    <div>{accountType}</div>
                    <div>{tradeAmount}</div>
                    <div>{currrency}</div>
                    <div>{PLID}</div>
                </div>
                <div className="fund__row2">
                    Share balance: {shareBalance}
                </div>
            </div>)
};


const VerifyTrade = (props) => {
    const { verifyTrade: {account, funds} } = props;
    const labels = verifyTradeLabels;

    return(
        <div className="verifyTrade">
            <div className="verifyTrade__account-details">
                <div>{account.name}</div>
                <div>{account.number}</div>
            </div>
            <div className="verifyTrade__labels">
                {createLabels(labels)}
            </div>
            <div className="verifyTrade__fund-details">
                {funds.map((fund)=> createFund(fund))}
            </div>
        </div>
    );
}


export default VerifyTrade;
