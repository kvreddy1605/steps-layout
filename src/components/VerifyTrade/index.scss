.verifyTrade{
    display: flex;
    flex-direction: column;
    padding: 0px 20px;

    &__account-details{
        width: 100%;
        padding-top: 15px;
        font-size: 20px;
    }

    &__labels{
        display: flex;
        border-bottom: 1px solid black;
        color: #B9B9B9;
        padding:15px 0px; 
        font-size: 14px;

        .label:first-child{
            width: 20%;
        }
        .label{
            width: 10%;
        }
    }

    &__fund-details{
        display: flex;
        flex-direction: column;
        font-size: 14px;

        .fund{
            display: flex;
            flex-direction: column;
            border-bottom: 1px solid black; 
        
            &__row1{
                display: flex;
                width: 100%;
                border-bottom: 1px solid #E1E1E1;
                padding: 15px 0px;

                div:first-child{
                    width: 20%;
                }
                div{
                    width: 10%;
                }
            }
            
            &__row2{
                padding: 5px 0px 15px 0px; 
            }
        }
    }
}