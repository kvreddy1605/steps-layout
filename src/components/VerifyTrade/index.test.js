import React from 'react';
import VerifyTrade from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('VerifyTrade Component Test', () => {

    it('SnapShot Test', () => {
        const verifyTrade = {
            account: {},
            funds: []
        }
        const wrapper = shallow(<VerifyTrade verifyTrade={verifyTrade} />) 
        expect(toJson(wrapper)).toMatchSnapshot();
    })
        
})

