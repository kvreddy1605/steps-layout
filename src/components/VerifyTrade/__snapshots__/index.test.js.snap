// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyTrade Component Test SnapShot Test 1`] = `
<div
  className="verifyTrade"
>
  <div
    className="verifyTrade__account-details"
  >
    <div />
    <div />
  </div>
  <div
    className="verifyTrade__labels"
  >
    <div
      className="label"
      key="0"
    >
      Description
    </div>
    <div
      className="label"
      key="1"
    >
      Ticker
    </div>
    <div
      className="label"
      key="2"
    >
      Firm/Branch
    </div>
    <div
      className="label"
      key="3"
    >
      Trade Date
    </div>
    <div
      className="label"
      key="4"
    >
      Trade Type
    </div>
    <div
      className="label"
      key="5"
    >
      Account Type
    </div>
    <div
      className="label"
      key="6"
    >
      Trade Amount
    </div>
    <div
      className="label"
      key="7"
    >
      Currrency
    </div>
    <div
      className="label"
      key="8"
    >
      PLID
    </div>
  </div>
  <div
    className="verifyTrade__fund-details"
  />
</div>
`;
